### PR TITLE
 Hack support IJ 2018.3

### PR DESCRIPTION
**Pull request by wcurrie**
Changes:
- Use newer IJ 2018.3. Maybe not required. I was confused by compile error for golang and scala.
- Use newer scala plugin. Uninformed hack because https://github.com/JetBrains/intellij-scala/commit/526787ba75 deleted `getTruncedQualifiedName`
- Copy/paste 2018.2 entires to 2018.3 in other languages: golang, javascript, python. Just aiming for "it compiles" and works for java.

### CODE REVIEW

File: java/src/com/google/idea/blaze/java/sync/importer/BlazeJavaWorkspaceImporter.java [hunk 1]
```diff
@@ -235,24 +235,22 @@ private void addTargetAsSource(
...
-        TargetIdeInfo depTarget = targetMap.get(dep.getTargetKey());
-        if (depTarget != null && Kind.JAVA_PROTO_LIBRARY_KINDS.contains(depTarget.getKind())) {
-          workspaceBuilder.directDeps.addAll(
-              depTarget.getDependencies().stream()
-                  .map(Dependency::getTargetKey)
-                  .collect(Collectors.toList()));
-        } else {
-          workspaceBuilder.directDeps.add(dep.getTargetKey());
-        }
+    // Add all deps always - workaround .jdeps missing external dependencies
```
Comment (wcurrie) [inline java/src/com/google/idea/blaze/java/sync/importer/BlazeJavaWorkspaceImporter.java:238]: Unrelated to 2018.3

File: scala/src/com/google/idea/blaze/scala/run/producers/BlazeScalaMainClassRunConfigurationProducer.java [hunk 2]
```diff
@@ -167,7 +167,7 @@ private static TargetIdeInfo getTarget(Project project, ScObject mainObject) {
     // Scala objects are basically singletons with a '$' appended to the class name.
     // The trunced qualified name removes the '$',
     // so it matches the main class specified in the scala_binary rule.
-    String qualifiedName = mainObject.getTruncedQualifiedName();
+    String qualifiedName = mainObject.qualifiedName();
```
Comment (wcurrie) [inline scala/src/com/google/idea/blaze/scala/run/producers/BlazeScalaMainClassRunConfigurationProducer.java:170]: This probably isn't the right replacement. Just aiming for: it compiles.

File: third_party/go/BUILD [hunk 3]
```diff
@@ -12,6 +12,8 @@ java_library(
         "intellij-ue-2018.1": ["@go_2018_1//:go"],
         "intellij-2018.2": ["@go_2018_2//:go"],
         "intellij-ue-2018.2": ["@go_2018_2//:go"],
+        "intellij-2018.3": ["@go_2018_2//:go"],
+        "intellij-ue-2018.3": ["@go_2018_2//:go"],
```
Comment (uri-canva) [inline third_party/go/BUILD:16]: You didn't update the third party references to the plugins, that's going to break support for those languages if the plugins are incompatible. I'm going to test some out and report back.

Comment (uri-canva) [reply third_party/go/BUILD:16]: Ok, at least for scala it still works, since we don't have any golang code yet in the main repo I think it's ok.

**Comment by uri-canva** [comment]
Cherry picked as https://github.com/uri-canva/intellij/commit/bc4a99f5355ccb68d7d8bba17e58bd3438842961 and released with https://github.com/uri-canva/intellij/commit/50e9edef2e58efa904869ccd5fb2f95f7a1902da.

File: intellij_platform_sdk/BUILD [hunk 4]
```diff
@@ -200,8 +200,8 @@ java_library(
 java_library(
     name = "jsr305",
     exports = select_for_ide(
-        clion = ["@jsr305_annotations//jar"],
-        default = [":plugin_api"],
+        default = ["@jsr305_annotations//jar"],
```
Comment (wcurrie) [inline intellij_platform_sdk/BUILD:203]: With 2018.3 it appears jsr305.jar is not longer bundled with the IJ distribution. Or maybe the name changed.

File: third_party/go/BUILD (hunk 3 above)
Comment (wcurrie) [reply third_party/go/BUILD:16]: Yup. I just ignored go. Should've explained sorry.

File: intellij_platform_sdk/BUILD (hunk 4 above)
Comment (uri-canva) [reply intellij_platform_sdk/BUILD:203]: The built plugin seems to be working so far, jsr305 is only a compile time annotation right?

**Comment by wcurrie** [comment]
Looks like this hack can be replaced with a rebase to get official support 🎉 https://github.com/bazelbuild/intellij/pull/432/files

**Pull request closed without merging**